Implement editor components with parameters

alexanderpann opened the pull request:
The parameter information is saved in the user object of the node instead of the root editor cell. The reason is that on first editor creation, this information can't be accessed.

For updated screenshots: see last answer

# Example

## Declaration

<img width="100" alt="image" src="https://github.com/JetBrains/MPS-extensions/assets/88385944/d14359eb-f3fa-46b2-b57a-7825ea3dd38e">

<img width="100" alt="image" src="https://github.com/JetBrains/MPS-extensions/assets/88385944/5136bd38-c4da-4d0f-b2f5-2e607b955965">

## Usage

<img width="100" alt="image" src="https://github.com/JetBrains/MPS-extensions/assets/88385944/6c37b58d-6943-4f99-8908-a02e2e6697ba">

kbirken commented:
Looks useful to be able to pass context information when calling the editor component. Just one proposal on syntax: It would look more consistent if the parameters were shown in the style of "overrides:" and "applicable concept:". I.e., just another section with the parameters:

```
  applicable concept:
    Interface
  parameters:
    node<> suppliedNode
    string name
```

BTW: Why is the first parameter in your example shown with type first, and the second parameter with the type after the colon?

alexanderpann commented:
I've implemented the feedback by @kbirken and Dennis Albrecht and also fixed the issue with the different parameter syntaxes. I like the way it looks now:

![](https://github.com/JetBrains/MPS-extensions/assets/88385944/47c0a502-07b3-4488-9e3f-4ed67b333c89)

![](https://github.com/JetBrains/MPS-extensions/assets/88385944/93e8375a-54ec-4203-a30a-d498211db555)
